### PR TITLE
Bug 1847583 - Fenix: Add search by bookmarks tests

### PR DESCRIPTION
**Pull request by sv-ohorvath**
Made verifyFirefoxSuggestResults  and verifySearchEngineSuggestionResults only one method as they were practically the same.
Added new tests based on TestRail manual tests.
Fixed defaultSearchCodeGoogleUS failing on the PR, for un-related reasons.
Ran 100x on Firebase, and all passed.

Discussed with Aaron how to handle code duplication in old and new Composed test classes, we've made a decision to freeze the old UI classes (to be deprecated in the future) and only add to the new Compose test classes.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1847583
https://bugzilla.mozilla.org/show_bug.cgi?id=1846941
https://bugzilla.mozilla.org/show_bug.cgi?id=1847774

### CODE REVIEW

**Review by AaronMT — APPROVED** [review]

File: fenix/app/src/androidTest/java/org/mozilla/fenix/ui/CreditCardAutofillTest.kt [hunk 1]
```diff
@@ -19,6 +19,7 @@ import org.mozilla.fenix.ui.robots.clickPageObject
 import org.mozilla.fenix.ui.robots.homeScreen
 import org.mozilla.fenix.ui.robots.navigationToolbar
 import java.time.LocalDate
+import org.junit.Ignore
```
[the remark this answers is not in the document]
Comment (sv-ohorvath) [reply fenix/app/src/androidTest/java/org/mozilla/fenix/ui/CreditCardAutofillTest.kt]: Yep, done.

**Comment by sv-ohorvath** [comment]
@Mergifyio backport releases_v117

**Comment by sv-ohorvath** [comment]
@Mergifyio backport releases_v116

**Comment by sv-ohorvath** [comment]
We actually don't want to uplift all the changes in this PR, so I'll create a new one.

**Pull request merged**